tests/tracer/test_http.py

```diff
@@ -48,60 +45,24 @@ def test_strip_query_string(url):
     )
 
 
-@given(urls())
-@example("")
-@example("/relative/path")
-@example("://?")
-@example("://a?b")
-@example("#fragment?with=query&string")
-@example(":")
-@example(":/")
-@example("://?&?")
-@example("://?&#")
+@pytest.mark.parametrize("url", _url_fixtures())
 def test_redact_url_obfuscation_disabled_without_param(url):
     assert redact_url(url, None, None) == url
 
 
-@given(urls())
-@example("")
-@example("/relative/path")
-@example("://?")
-@example("://a?b")
-@example("#fragment?with=query&string")
-@example(":")
-@example(":/")
-@example("://?&?")
-@example("://?&#")
+@pytest.mark.parametrize("url", _url_fixtures())
 def test_redact_url_obfuscation_disabled_with_param(url):
     assert redact_url(url, None, "query_string") == url
 
 
-@given(urls())
-@example("")
-@example("/relative/path")
-@example("://?")
-@example("://a?b")
-@example("#fragment?with=query&string")
-@example(":")
-@example(":/")
-@example("://?&?")
-@example("://?&#")
+@pytest.mark.parametrize("url", _url_fixtures())
 def test_redact_url_not_redacts_without_param(url):
     res = redact_url(url, re.compile(b"\\@"), None)
     expected_result = url if isinstance(res, str) else url.encode("utf-8")
     assert res == expected_result
 
 
-@given(urls())
-@example("")
-@example("/relative/path")
-@example("://?")
-@example("://a?b")
-@example("#fragment?with=query&string")
-@example(":")
-@example(":/")
-@example("://?&?")
-@example("://?&")
```

Comment:
Note: this test case had one different example compared to the other tests: `://?&` compared to `://?&#`. Trying to add the latter resulted in this test failing, so I left this one out for the time being. @gnufede I noticed this was added in your previous change #3980, can you give me some context as to why this test used a different example and why `://?&#` causes the test to fail?

Reply:
Update: `urllib.parse.urlparse` doesn't recognize empty fragment strings, so `urllib.parse.urlunparse` doesn't create the fragment identifier string `#` in this edge case. I've changed the example to include a non-empty fragment string instead.